upload-cloud-storage/README.md

```diff
@@ -0,0 +1,194 @@
+<!--
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# upload-cloud-storage
+
+This action uploads files/folders to a [Google Cloud Storage (GCS)][gcs] bucket. This is useful when
+you want upload build artifacts from your workflow.
+
+Paths to files that are successfully uploaded are set as output variables and can be
+used in subsequent steps.
+
+## Prerequisites
+
+- This action requires Google Cloud credentials that are authorized to upload
+  blobs to the specified bucket. See the Authorization section below for more
+  information.
+
+## Usage
+
+### For uploading a file
+
+```yaml
+steps:
+  - id: upload-file
+    uses: GoogleCloudPlatform/github-actions/upload-cloud-storage@master
+    with:
+      path: /path/to/file
+      destination: bucket-name/file
+
+  # Example of using the output
+  - id: uploaded-files
+    uses: foo/bar@master
+    env:
+      file: ${{steps.upload-file.outputs.uploaded}}
+```
+
+The file will be uploaded to `gs://bucket-name/file`
+
+### For uploading a folder
+
+```yaml
+steps:
+  - id: upload-files
+    uses: GoogleCloudPlatform/github-actions/upload-cloud-storage@master
+    with:
+      path: /path/to/folder
+      destination: bucket-name
+
+  # Example of using the output
+  - id: uploaded-files
+    uses: foo/bar@master
+    env:
+      files: ${{steps.upload-files.outputs.uploaded}}
+```
+
+If the folder has the following structure:
+
+```
+.
+└── folder
+    ├── file1
+    └── folder2
+        └── file2
+```
+
+The files will be uploaded to `gs://bucket-name/folder/file1`,`gs://bucket-name/folder/folder2/file2`
+
+## Inputs
+
+- `path`: (Required) The path to a file or folder inside the action's filesystem
```

Comment:
Something to consider for a future enhancement, allowing file globs in the path.

Reply:
GCS library requires the fully qualified path to the file, but we can support this in the future by having our own helpers to evaluate globs, similar to how we have the directory walk helper.